CMakeLists.txt

```diff
@@ -131,10 +131,12 @@ include("cmake/host_compiler.cmake")
 include("cmake/configuring_primitive_list.cmake")
 
 if(UNIX OR MINGW)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
+    if (NOT CMAKE_C_STANDARD)
+        set(CMAKE_C_STANDARD 99)
```

Comment:
It looks like CMake 2.8.12 doesn't support `CMAKE_C_STANDARD` and `CMAKE_CXX_STANDARD`. We claim to support CMake 2.8.12 and newer.

Reply:
That release is from 2013. How relevant is it still for you?

Also, you should really consider upgrading to C++14, because of Xbyak.

Reply:
We have minimal requirements such as CMake 2.8.12 and C++11 because RHEL 7 is distributed with CMake 2.8.12 and GNU 4.8 that supports C++11 (except some features i.e. regex). We want users to be able to build oneDNN on RHEL 7 without installing additional software.

Reply:
They can install devtoolset-8, 9 or 10 and get GCC 8, 9, or 10, which support C++17. EPEL 7 will also give you CMake 3.17.

But we don't need to do that. We just need to compile oneDNN with C++14 or 17 where the compiler does support it. If you choose to use an old compiler, you get worse code, but the rest of us who have modern environments don't pay that penalty.

I'll update this patch to detect the CMake version.

Reply:
I think it's reasonable to provide a way to build oneDNN with different standards.